engine/consensus/compliance.go

```diff
@@ -0,0 +1,20 @@
+package consensus
+
+import (
+	"github.com/onflow/flow-go/model/events"
+	"github.com/onflow/flow-go/model/messages"
+)
+
+// ProposalProvider provides proposals created by this node to non-consensus nodes.
+type ProposalProvider interface {
+	// ProvideProposal asynchronously submits our proposal to all non-consensus nodes.
+	ProvideProposal(proposal *messages.BlockProposal)
+}
+
+// ComplianceProcessor accepts blocks for ingestion and processing by the compliance layer.
+// It is used to ingest blocks received through other means than the HotStuff happy path,
+// like the block synchronization engine.
+type ComplianceProcessor interface {
```

Comment:
Conceptually our engines deal with two types of inputs: inputs from other replicas and locally generated data. While locally generated data is trusted, inputs from other nodes aren't, they are subject to byzantine behavior. 
Based on specification of: [MessageProcessor](https://github.com/onflow/flow-go/blob/feature/active-pacemaker/network/engine.go#L47) this interface is responsible for handling such inputs. I would like to keep that convention and submit everything from other replicas using `Process` and use custom ingestion logic for dealing with locally generated events. Currently we are violating this "contract" by submitting unchecked inputs using other means and not `Process`.

Reply:
Good point, for this case I'll change to `MessageProcessor`